src/main/java/org/apache/sysml/runtime/matrix/data/SparseBlockCSR.java

```diff
@@ -848,7 +848,57 @@ public String toString() {
 		
 		return sb.toString();
 	}
-	
+
+//	@Override
+	public boolean checkValidity(int rlen, int clen, boolean strict) {
+		//1. correct meta data
+		if( rlen < 0 || clen < 0 ) {
+			throw new RuntimeException("Invalid block dimensions: "+rlen+" "+clen);
```

Comment:
This was a good reminder, to double check the sparse block meta data which already revealed a bad merge.

Regarding the meta data check, I would recommend to extend the api to also pass long nnz as an argument. The checks should then validate: 
(a) that _ptr.length > rlen. In CSR the pointer array is supposed to be of length rlen+1. However, due to reuse of sparse blocks, the actual rlen can be smaller in which case we simply ignore the too long _ptr array.
(b) that _indexes.length >= nnz, _values.length >= nnz, and size==nnz. Similar to java collections, the underlying indexes and values arrays can be larger to allow for efficient allocation (exponential resizing).